tests/publiccloud/patch_and_reboot.pm

```diff
@@ -27,6 +28,10 @@ sub run {
     $args->{my_instance}->softreboot();
 }
 
+sub post_fail_hook {
+    ssh_interactive_post_fail();
+}
+
```

Comment:
I think we need to call the `post_fail_hook()` from consoletest, or we might miss something there. WDYT?
```
sub post_fail_hook {
    $self = shift;
    ssh_interactive_post_fail();
    $self->SUPER::post_fail_hook;
}
```

Reply:
Same with all the others...

Reply:
I don't think so because on this phase we run everything like `ssh instance zypper dup` so the standard consoletest post fail hook does not work for us.

The only 'problem' is see is in `patch_and_reboot.pm` where we are upgrading the instance and not collecting the logs from it. But I would keep this as a future improvement.